gateway/src/services/config-manager-v2.ts

```diff
@@ -281,17 +358,31 @@ export class ConfigManagerV2 {
     }
 
     // Extract the namespace ids.
-    const namespaceMap: ConfigurationRoot = {};
-    for (const namespaceKey of Object.keys(configRoot)) {
+    const namespaceMap: ConfigurationPattern = {};
+    for (const namespaceKey of Object.keys(configRoot.configurations)) {
       namespaceMap[namespaceKey.slice(NamespaceTag.length)] =
-        configRoot[namespaceKey];
+        configRoot.configurations[namespaceKey];
     }
 
-    // Rebase the file paths in config root if they're relative paths.
+    // Rebase the file paths in config & template roots if they're relative paths.
     for (const namespaceDefinition of Object.values(namespaceMap)) {
       for (const [key, filePath] of Object.entries(namespaceDefinition)) {
-        if (filePath.charAt(0) !== '/') {
+        if (!path.isAbsolute(filePath)) {
           namespaceDefinition[key] = path.join(configRootDir, filePath);
+          if (key === 'configurationPath') {
+            namespaceDefinition['templatePath'] = path.join(
+              ConfigTemplatesDir,
+              filePath
+            );
+          }
+        } else {
+          //assumption is made that config file is relative to root config file
+          if (key === 'configurationPath') {
+            namespaceDefinition['templatePath'] = path.join(
+              ConfigTemplatesDir,
+              path.relative(configRootDir, filePath)
```

Comment:
If the file path is absolute, and yet we want to map it to a template - then this mapping doesn't make sense anymore. We should simply raise an error instead and disallow any absolute paths in the configuration definitions.